db/db.go

```diff
@@ -0,0 +1,131 @@
+package db
+
+import (
+	"context"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// DB database
+type DB struct {
+	pool *pgxpool.Pool
+	psql sq.StatementBuilderType
+}
+
+// New creates a new DB connection
+func New(pool *pgxpool.Pool, psql sq.StatementBuilderType) *DB {
+	return &DB{
+		pool: pool,
+		psql: psql,
+	}
+}
+
+// Post ...
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts() ([]Post, error) {
+	posts := []Post{}
```

Comment:
I have a question about line 35. When I used `var posts []Post{}`, and scanned the result of the database to it, I got `null` instead of an empty array. I don't understand why that happens.